test/asynchronous/test_dns.py

```diff
@@ -264,6 +267,12 @@ async def test_3_throw_when_return_address_is_identical_to_srv_hostname(self):
                 "mock_target": "mongo.local",
                 "expected_error": "Invalid SRV host",
             },
+            # When the SRV hostname has three or more dot-separated parts
+            # it is valid for the returned hostnames to be identical.
+            {
+                "query": "_mongodb._tcp.blogs.mongodb.com",
+                "mock_target": "blogs.mongodb.com",
+            },
```

Comment:
I suggest moving this to a new test since we're testing success here and the test name is "test_3_throw_when_return_address_is_identical_to_srv_hostname".